profile/src/wallet_kit_common/types/logged_result.rs

```diff
@@ -0,0 +1,54 @@
+use crate::prelude::*;
+
+pub trait LoggedResult<T: SafeToLog>: Sized {
+    fn log_lvl_formatted<F>(self, level: log::Level, format: F) -> Self
+    where
+        F: FnOnce(&T) -> String;
+
+    fn log_lvl(self, level: log::Level, prefix: impl AsRef<str>) -> Self {
+        self.log_lvl_formatted(level, |f| {
+            format!("{} - {:?}", prefix.as_ref(), f.non_sensitive())
+        })
+    }
+    fn log_info(self, prefix: impl AsRef<str>) -> Self {
+        self.log_lvl(log::Level::Info, prefix)
+    }
+}
+
+impl<T: SafeToLog> LoggedResult<T> for Result<T> {
```

Comment:
not yet heavily used, idea is code ought to prefer functional chaining, over `?`, and the idea here is to be able to easily log Ok value and Err.

Reply:
to SAFELY log that is.